src/Microsoft.DotNet.Interactive.Tests/Server/StandardIOKernelServerTests.cs

```diff
@@ -198,6 +198,31 @@ public async Task it_produces_values_when_executing_Console_output()
             stdOut.Should().Contain(guid);
         }
 
+        [Fact]
+        public async Task missing_command_returns_a_diagnostic_event()
+        {
+            // since it's not a valid command we have to manually create it
+            var command = new
+            {
+                CommandType = "ThisIsNotAValidCommandType",
+                Command = new object(), // body doesn't matter
+            };
+            var writer = new StringWriter();
+            Serializer.JsonSerializer.Serialize(writer, command);
+            var commandText = writer.ToString();
+
+            await _standardIOKernelServer.WriteAsync(commandText);
+
+            _kernelEvents
+                .Should()
+                .ContainSingle<KernelEventEnvelope<DiagnosticLogEntryProduced>>()
```

Comment:
I think this should be at least a commandFailed event @jonsequitur  what are your thoughts?

Reply:
We can't do command failed event, because that requires returning the command that failed, so then on the other side, the client would try to deserialize another invalid command object:

``` json
{
  "eventType": "CommandFailed",
  "event": "...",
  "command": {
    "commandType": "NotACommand",
   "...": "this will fail to parse according to `contracts.ts`",
  }
}
```

Reply:
we can roundtrip the evnelope

Reply:
this is a logic that does not make it to the command pipeline, is done at the transport layer when commands are deserialised

Reply:
A diagnostic event isn't forceful enough. 

One possibility is the command on the `CommandFailed` event could be null. What do we do when there's a serialization error?

Reply:
I've been experimenting and we can return a `CommandFailed` event with a null `command` field, but since that field is `null` there's no token, so we can't pair the failed event with the client-side command that triggered it.  The end result is that the client never knows that a `CommandFailed` event comes back so it's forever waiting for a result.

It seems like we have 2 options at this point:

1. Populate a fake/shim `command` field so we can regurgitate the token so the client get notified that something terrible happened, or
2. Drop this PR since we now tightly couple the kernel server and client extension, so we're much less likely to get into a situation where the client submits a command that it's not prepared to handle.

Reply:
I think #2 is fine. This is a programming error, so handling it within the domain doesn't make a lot of sense.